src/app/frontend/login/component.ts

```diff
@@ -36,7 +37,7 @@ enum LoginModes {
 })
 export class LoginComponent implements OnInit {
   loginModes = LoginModes;
-  selectedAuthenticationMode = LoginModes.Kubeconfig;
+  selectedAuthenticationMode = this.cookies_.get('selectedAuthenticationMode') || '';
```

Comment:
Do not hardcode the name. Add it to the [config](https://github.com/kubernetes/dashboard/blob/master/src/app/frontend/index.config.ts) and inject into the constructor.
```ts
@Inject(CONFIG_DI_TOKEN) private readonly config_: Config,
```

Reply:
done